src/daft-logical-plan/src/ops/join.rs

```diff
@@ -314,3 +258,39 @@ impl Join {
         res
     }
 }
+
+#[cfg_attr(feature = "python", pyclass)]
+#[derive(Clone, TypedBuilder, Default)]
+#[builder(field_defaults(setter(into)))]
+pub struct JoinColumnRenamingParams {
```

Comment:
this is a bit of a mouthful, what about renaming this to just `JoinOptions`. That'd leave more flexibility for putting additional options in there as well.